### DISCUSSION TITLE
User Filter by Logging Station Callsign

### DISCUSSION THREAD

**Discussion by vazquezrodjoel** [foldynl/QLog#524] [Q&A]
Hello,
Is there a way to make a User Filter by Logging Station Callsign but by specific callsign rather than just blank or not?  I want to filter my contacts by my previous callsigns.  Blank is also useful in some cases for log correction when needed, but cant find a way to filter my specific callsigns.

**Comment by aa5sh** [comment]
Not sure if it is something off with my database but I think there may be a
bug in the user filters.  If I edit one of my filters the field names are
wrong from what I had them in 39.  If I change them to be correct, get out
and back in they are wrong again.

If I load the code and just display the field list in
logbookmodel::getAllFieldNamesTranslation() with a qWarning() << ret;
(sorry I am horrible at debugging at this point in QT), each time I run it
the list changes and it isn't in the proper order.

**Comment by aa5sh** [comment]
I think I found the issue.  I will put it in a Pull request but for
fieldNameTranslationMap   was defined as QHash instead of QMap.  From the
docs a QHash will in a arbitrary order whereas QMap should be in a
consistent order.

Michael, AA5SH
